src/lio/warmer_helpers.h

```diff
@@ -0,0 +1,220 @@
+/*
+   Copyright 2016 Vanderbilt University
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+//***********************************************************************
+// Warmer helper definitions
+//***********************************************************************
+
+
+#ifndef _WARMER_H_
+#define _WARMER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define WFE_SUCCESS   1
+#define WFE_FAIL      2
+#define WFE_WRITE_ERR 4
+
+typedef struct {
+    int  state;
+    int  nfailed;
+    char name[1];
+} __attribute__((packed)) db_inode_entry_t;
```

Comment:
I assume that precisely because that's what we do now :)

It's not a performance argument, it's a portability argument.

If I understand you properly, the justification for packing the struct is "I want to be able to serialize and later deserialize between arbitrary combinations of (compiler, architecture)". If that's the case, the solution becomes a lot more complex than "tack a __packed__" on there (see: solutions like protobuf and similar). Ignoring the fact that misaligned reads on (ex) x86 are slow (which I wasn't concerned about), my concern is that on some arches, misaligned accesses are either unsupported in hardware or require different instructions. If you're not careful to supply the right compiler flags and hints to the code, you'll get a bus fault.

Outside of all that, the types "int" and "char" aren't required to be a certain number of bytes. Dropping the padding won't save you if you aren't consistently trying to read the right number of bytes out :) (I don't care one bit about big-endian arches or weirdos where `CHAR_BIT != 8`)